.github/workflows/static-analysis.yml

```diff
@@ -2,6 +2,9 @@ name: static analysis
 
 on:
   push:
+    branches:
+      - master
+      - '*.x'
   pull_request:
   schedule:
     - cron: '0 0 * * *'
```

Comment:
Btw, maybe also remove the cron check here. I don't think it's important to let that one run regularly.

Reply:
I have removed the static analysis cron, but will leave this review open, so that other people can see it.